rust/gui-client/src-tauri/src/client.rs

```diff
@@ -65,6 +65,10 @@ pub(crate) fn run() -> Result<()> {
     std::panic::set_hook(Box::new(tracing_panic::panic_hook));
     let cli = Cli::parse();
 
+    rustls::crypto::ring::default_provider()
+        .install_default()
+        .expect("Calling `install_default` only once per process always succeeds");
```

Comment:
@ReactorScram Is this a good place? I was tempted to put this into `main.rs` to ensure it will definitely only ever be called once per process. I couldn't find any other callers of `run` so I guess it should be fine. I am not sure why we extracted `run` instead of putting this stuff into `main` but if it is for reusability reasons, then we should probably move this to `main` instead.

Thoughts?

Reply:
Yeah that should work. The `version-check` CI job used to run on macOS, where `gui-client` can't compile, so some of the strange designs are made to turn the whole package into a stub on macOS.
```suggestion
    rustls::crypto::ring::default_provider()
        .install_default()
        .expect("Calling `install_default` only once per process should always succeed");
```